spec/services/approve_appeal_service_spec.rb

```diff
@@ -4,27 +4,55 @@
 
 RSpec.describe ApproveAppealService do
   describe '#call' do
-    context 'with an existing appeal' do
-      let(:appeal) { Fabricate(:appeal) }
-      let(:account) { Fabricate(:account) }
-
-      it 'processes the appeal approval' do
-        expect { subject.call(appeal, account) }
-          .to mark_overruled
-          .and record_approver
-      end
+    let(:appeal) { Fabricate(:appeal) }
+    let(:account) { Fabricate(:account) }
+
+    it 'processes the appeal approval' do
+      expect { subject.call(appeal, account) }
+        .to mark_overruled
+        .and record_approver
+    end
 
-      def mark_overruled
-        change(appeal.strike, :overruled_at)
-          .from(nil)
-          .to(be > 1.minute.ago)
+    context 'with an appeal about then-deleted posts marked as sensitive by moderators' do
+      let(:target_account) { Fabricate(:account) }
+      let(:appeal) { Fabricate(:appeal, strike: strike, account: target_account) }
+      let(:deleted_media) { Fabricate(:media_attachment, type: :video, status: Fabricate(:status, account: target_account), account: target_account) }
+      let(:kept_media) { Fabricate(:media_attachment, type: :video, status: Fabricate(:status, account: target_account), account: target_account) }
+      let(:strike) { Fabricate(:account_warning, target_account: target_account, action: :mark_statuses_as_sensitive, status_ids: [deleted_media.status.id, kept_media.status.id]) }
+
+      before do
+        target_account.unsuspend!
+        deleted_media.status.discard!
       end
 
-      def record_approver
-        change(appeal, :approved_by_account)
-          .from(nil)
-          .to(account)
+      it 'redirects back to the strike page and notifies target account about approved appeal', :inline_jobs do
```

Comment:
Tiny thing, but since you moved this out of the controller spec, "redirects" are not a thing here anymore.

Maybe something like

```suggestion
      it 'approves the appeal, marks the statuses as not sensitive and notifies target account about the approval', :inline_jobs do
```